optimum/intel/generation/modeling.py

```diff
@@ -296,8 +295,24 @@ def forward(
 
             inputs["past_key_values"] = past_key_values
 
-        if position_ids is not None and model_type in MODEL_TYPES_REQUIRING_POSITION_IDS:
-            inputs["position_ids"] = position_ids
+        if has_position_ids and position_ids is not None:
+            inputs.update({"position_ids": position_ids})
+        elif has_position_ids and position_ids is None:
+            seq_length = input_ids.shape[-1]
+            if not self.use_cache:
+                past_key_values_length = 0
+            else:
+                past_key_values_length = (
+                    past_key_values[0].shape[-2]
+                    if model_type.replace("-", "_") in MULTI_QUERY_ATTN_MODELS
+                    else past_key_values[0][1].shape[-2]
+                )
+            position_ids = torch.arange(
+                past_key_values_length, seq_length + past_key_values_length, dtype=torch.long, device=self._device
+            ).unsqueeze(0)
+            inputs.update({"position_ids": position_ids})
+        elif not has_position_ids and position_ids is not None:
+            logger.warning("You miss the position_ids in the inputs")

```

Comment:
I think it would be better to check directly in the model graph to check whether `position_ids` is one of the model's expected input if that can be done. If not this new addition will create issues for all the previously exported models (the ones that were exported without any `position_ids)for all architectures from `MODEL_TYPES_REQUIRING_POSITION_IDS`
```suggestion
        if "position_ids" in self.input_names:
            if position_ids is None:
                position_ids = ...
            inputs["position_ids"] = position_ids
```

Reply:
also concerning the `position_ids` (and the `attention_mask`) computation I think we should do as follow : https://github.com/huggingface/optimum-intel/blob/819e3e824bdcef574a1487f700acd9a2fe39ddac/optimum/intel/openvino/modeling_decoder.py#L365-L383

Reply:
I am afraid there is no `input_names` attr in this class.

Reply:
yes could this be added you think ?

Reply:
Yes, I can try it